flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -154,8 +154,11 @@
 	/** This flag defines if we use compression for the state snapshot data or not. Default: false */
 	private boolean useSnapshotCompression = false;
 
-	/** Determines if a task fails or not if there is an error in writing its checkpoint data. Default: true */
-	private boolean failTaskOnCheckpointError = true;
+	/**
+	 * @deprecated Should no longer be used because we would not support to let task fail on checkpoint error.
```

Comment:
I think we should mention the (kind of) replacement flag for tolerable checkpoint errors here. And also how the two interact.